pkg/network/nmstate.go

```diff
@@ -51,9 +51,23 @@ func renderNMState(conf *cnao.NetworkAddonsConfigSpec, manifestDir string, clust
 		return nil, errors.Wrap(err, "failed to render nmstate state handler manifests")
 	}
 
+	if clusterInfo.NmstateOperator {
+		if len(objs) == 0 {
+			return nil, fmt.Errorf("no nmstate custom resource found")
+		}
+		parametrizeNMStateCR(objs[0], conf)
+	}
 	return objs, nil
 }
 
+func parametrizeNMStateCR(nm *unstructured.Unstructured, conf *cnao.NetworkAddonsConfigSpec) {
```

Comment:
Can't we just render the same way we do [here](https://github.com/kubevirt/cluster-network-addons-operator/pull/1005/commits/f750bca9de9a2d33f105ce54e79b1f974b043f63#diff-efe99982594a52b1566407a711ce4c2eaf55c6ba33f9e9911b6d305a69350fe9R40)?

Reply:
We would need to create a special yaml for that in kubernetes-nmstate I think, because when cnao copies kubernetes-nmstate folder, we diff the NMState CR in CNAO and in kubernetes-nmstate.

Reply:
I don't follow... o_0
+ we currently don't deploy knmstate operator, but when I "hacked" it to deploy it, I'm getting this error
```
  - lastHeartbeatTime: "2021-09-09T15:40:15Z"
    lastTransitionTime: "2021-09-09T15:39:08Z"
    message: 'could not apply (nmstate.io/v1beta1, Kind=NMState) /nmstate: could not
      retrieve existing (nmstate.io/v1beta1, Kind=NMState) /nmstate: no matches for
      kind "NMState" in version "nmstate.io/v1beta1"'
    reason: FailedToApply
    status: "True"
    type: Degraded
```
am I doing something wrong?

Reply:
CNAO will use NMState operator if the operator is installed (normally openshift + the openshift NMState operator) this way we can have CNV + NMState operator working fine.

The error is because the NMState CRD is no deployed since the knmstate operator has not being deployed either, maybe we need to put some CI for it ?

Reply:
I think we have some basic tests for this:
The error is because the NMState CRD is no deployed since the knmstate operator has not being deployed either, maybe we need to put some CI for it ?

What is not being tested (or done at the moment) is that when an nmstate-operator is installed, and we add nmstate to thte cluster-network-addons-config, CNAO should also pass nodeSelector from its PlacementConfiguration, which is currently not passed IIUC.